state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2053,20 +2068,45 @@ pub mod pallet {
 			})
 		}
 
-		pub(super) fn take_network_fee(input: AssetAmount, min_fee_enforced: bool) -> FeeTaken {
+		/// Take network fee from the input amount and return the remaining amount and the fee.
+		/// If `swap_request_id` is provided, the minimum network fee is enforced.
+		pub(super) fn take_network_fee(
+			input: AssetAmount,
+			swap_request_id: Option<SwapRequestId>,
+		) -> FeeTaken {
 			if input.is_zero() {
 				return FeeTaken { remaining_amount: 0, fee: 0 };
 			}
 
-			let min_fee = if min_fee_enforced { MinimumNetworkFeePerChunk::<T>::get() } else { 0 };
+			// Check if minimum network fee still needs to be filled for this swap request.
+			let network_fee = T::NetworkFee::get();
+			let fee = if let Some(swap_request_id) = swap_request_id {
+				let swap_request =
+					SwapRequests::<T>::get(swap_request_id).expect("Swap request should exist");
```

Comment:
I think if it is not locally clear that this won't panic (like if you just checked it a line above), we should wirte code more defensively: we could simply fallback to charging 0 if you can't find the information you need.